### PR TITLE
Enable MGTwoLevelTransfer::reinit_geometric_transfer for FECollections

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

File: include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h [hunk 1]
```diff
@@ -869,14 +869,49 @@ namespace internal
...
+              std::max(max_active_cell_indicices[1], cell->active_fe_index());
+          }
+
+      const auto comm = get_mpi_comm(dof_handler_fine);
+
+      ArrayView<unsigned int> temp_min(min_active_cell_indicices);
+      ArrayView<unsigned int> temp_max(max_active_cell_indicices);
+      Utilities::MPI::min(temp_min, comm, temp_min);
+      Utilities::MPI::max(temp_max, comm, temp_max);
+
```
Comment (marcfehling) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h:900]:
```suggestion

      // make sure that no hp is used on either coarse or fine dofhandler respectively
```

**Review by marcfehling — CHANGES_REQUESTED** [review]
The assertions state that this is only available if all elements in a `FECollection` are identical, and all cells on each `DoFHandler` have the same active FE index. Right?

**Comment by peterrum** [comment]
Thanks for the review.  I have made the changes. Regading:

> The assertions state that this is only available if all elements in a FECollection are identical

No.

> and all cells on each DoFHandler have the same active FE index. Right?

Yes.

File: include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h [hunk 2]
```diff
@@ -868,15 +868,52 @@ namespace internal
       // copy constrain matrix; TODO: why only for the coarse level?
       transfer.constraint_coarse.copy_from(constraint_coarse);
 
-      // make sure that no hp is used
-      AssertDimension(dof_handler_coarse.get_fe_collection().size(), 1);
-      AssertDimension(dof_handler_fine.get_fe_collection().size(), 1);
+      // gather ranges for active FE indices on both fine and coarse dofhandlers
+      std::array<unsigned int, 2> min_active_fe_indicices = {
```
Comment (kronbichler) [inline include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h:872]:
```suggestion
      std::array<unsigned int, 2> min_active_fe_indices = {
```
(and the same for `max_fe_indices`).

**Review by kronbichler — APPROVED** [review]
Looks good to me apart from a typo in a variable name.

File: include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h (hunk 2 above)
Comment (peterrum) [reply include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h]: Fixed.

**Pull request merged**